May news (#1051)
ropensci/roweb3: content/blog/2025-05-26-news-may-2025/index.Rmd

- Vignette, knitr eval option:   https://github.com/ropensci/gtexr/blob/592ac781672f07eb67e935d4155570c5960d1fdb/vignettes/gtexr.Rmd#L16

### More on {renv} and R versions

To follow up on last newsletter's item ["🔗
CI: Pin the R version if using renv"](/blog/2025/04/28/news-april-2025/#ci-pin-the-r-version-if-using-renv), a tip by Hugo Gruson:
it's also possible to automatically set the R version to the one documented in `renv.lock`
when using the [`r-lib/setup-r`action](https://github.com/r-lib/actions/blob/4f58f8ffa872cbd4cb016c51ec497fdbe0a02711/setup-r/README.md?plain=1#L37).

### `usethis::use_air()`

In March we mentioned [Air](/blog/2025/03/27/news-march-2025/#air-a-new-formatter-for-r), the new formatter developed at Posit.
In usethis development version, there's a function that helps you set up a project to use Air: [`use_air()`](https://usethis.r-lib.org/dev/reference/use_air.html).
See also the [issue about the JSON config](https://github.com/r-lib/usethis/issues/2114).

### Help us maintain {pkgcheck}!

> You might know of [our {pkgcheck} system](https://docs.ropensci.org/pkgcheck/) used for our automated package checks, and for your own stand-alone and GitHub-action usage on any package. 

Reviewer: ```suggestion
You might know of [our {pkgcheck} system](https://docs.ropensci.org/pkgcheck/) used for our automated package checks, and for your own stand-alone and [GitHub-action usage](https://github.com/ropensci-review-tools/pkgcheck-action) on any package. 
```

We are seeking co-maintainers for the {pkgcheck} package that powers that. 

> In particular, we have several ideas for additional checks in current repo issues. 

Reviewer: ```suggestion
In particular, we have several [ideas for additional checks in current repo issues](https://github.com/ropensci-review-tools/pkgcheck/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22%3Amag%3A%20Check%20Idea%22). 
```

We would like volunteers to attend an online get-together for us to coach you on how to implement new checks, and so to help you get involved in maintaining and further developing our checking system.

> Please email `mark@ropensci.org` if you're interested!

Reviewer: ```suggestion
If you're interested and already part of our slack group, please join our dedicatated `#pkgcheck` channel, otherwise email `mark@ropensci.org`.
```

  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(trimws(...), sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### Help us maintain {pkgcheck}!

You might know of [our {pkgcheck} system](https://docs.ropensci.org/pkgcheck/) used for our automated package checks, and for your own stand-alone and [GitHub-action usage](https://github.com/ropensci-review-tools/pkgcheck-action) on any package. 
We are seeking co-maintainers for the {pkgcheck} package that powers that. 
In particular, we have several [ideas for additional checks in current repo issues](https://github.com/ropensci-review-tools/pkgcheck/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22%3Amag%3A%20Check%20Idea%22). 
We would like volunteers to attend an online get-together for us to coach you on how to implement new checks, and so to help you get involved in maintaining and further developing our checking system.

> If you're interested and already part of our slack group, please join our dedicated `#pkgcheck` channel, otherwise email `mark@ropensci.org`.

Reviewer 2: ```suggestion
If you're interested and already part of the rOpenSci Slack, please join our dedicated `#pkgcheck` channel, otherwise email `mark@ropensci.org`.
```

> 

Reviewer 3: ```suggestion
## Champions Program 2025

The call for the first cohort in Spanish has closed.  We received 121 aplications from people in 22 different countries. 
Most of this year's applicants want to develop a new package and the topics are very varied: from data access, to new analytics, to educational materials and health and agricultural applications.
The review process is in full swing. Notifications will be going out over the next few weeks to all those who applied. 
We greatly appreciate the time all applicants took to write and submit their proposals. 
```

Reviewer 2: suggestions: "educational materials, to health and agricultural applications." (add comma and 'to')

And perhaps an enthousiastic sign off sentence? Something like "We can't wait to see what this year's cohort will bring to the program!"

### Coworking 

Read [all about coworking](/blog/2023/06/21/coworking/)!

> - Tuesday Jun 3, 2025 2:00 PM CEST (12:00 UTC), ["Getting to know the DSLC"](/events/coworking-2025-06/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Jon Harmon](/author/jon-harmon/).
>     - Visit the [DSLC](https://dslc.io/).
> 

Reviewer 2: @yabellini Did you have some ideas for the July 1st coworking? No hurry and no worries if not 😁

```suggestion
- Tuesday Jun 3, 2025 2:00 PM CEST (12:00 UTC), ["Getting to know the Data Science Learning Community"](/events/coworking-2025-06/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Jon Harmon](/author/jon-harmon/).
    - Visit the [DSLC](https://dslc.io/)
    - Learn more about the DSLC and how you might get involved
- Tuesday July 1, 2025 9:00 AM Americas Pacific (16:00 UTC), "TBA" with [Yanina Bellini Saibene](/author/yanina-bellini-saibene) and cohost TBA.

```

Reviewer 3: Yes!! Saranjeet Kaur on "Research Software Engineer and R". She confirmed.

Reviewer 2: Hooray! I'll try to get the event posted today, at least a draft PR that @maelle could merge when publishing the Newsletter. Saranjeet gets a chance to review first, great, otherwise she can tweak the details whenever she gets the chance.

  
  string
}
```

```{r, results='asis'}
software_review <- posts[purrr::map_lgl(posts, ~any(.x[["software_peer_review"]]))]
if (length(software_review) > 0) {
  cat("### Software Review\n\n")
  cat(
    paste0(
      purrr::map_chr(software_review, format_post),
      collapse = "\n\n"
    )
  )
  cat("\n\n")
}

others <- posts[purrr::map_lgl(posts, ~any(.x[["other"]]))]
if (length(others) > 0) {
  cat(
    paste0(
      purrr::map_chr(others, format_post),
      collapse = "\n\n"
    )
  )
  cat("\n\n")
}

tech_notes <- posts[purrr::map_lgl(posts, ~any(.x[["tech_note"]]))]
if (length(tech_notes) > 0) {
  cat("\n\n")
  cat("### Tech Notes\n\n")
  cat(
    paste0(
      purrr::map_chr(tech_notes, format_post),
      collapse = "\n\n"
    )
  )
  cat("\n\n")
}
```

## Calls for contributions

### Calls for maintainers

If you're interested in maintaining any of the R packages below, you might enjoy reading our blog post [What Does It Mean to Maintain a Package?](/blog/2023/02/07/what-does-it-mean-to-maintain-a-package/).

* [hddtools](https://docs.ropensci.org/hddtools/), Hydrological Data Discovery Tools. [Issue for volunteering](https://github.com/ropensci/hddtools/issues/36).

> * [USAboundaries](https://docs.ropensci.org/USAboundaries/) (and USAboundariesdata), historical and contemporary boundaries of the United States of America . [Issue for volunteering](https://github.com/ropensci/USAboundaries/issues/50).

Reviewer 2: ```suggestion
* [USAboundaries](https://docs.ropensci.org/USAboundaries/) (and USAboundariesdata), historical and contemporary boundaries of the United States of America. [Issue for volunteering](https://github.com/ropensci/USAboundaries/issues/50).
```

* [historydata](https://docs.ropensci.org/historydata/), datasets for historians. [Issue for volunteering](https://github.com/ropensci/historydata/issues/23).

### Calls for contributions

Refer to our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

## Package development corner

Some useful tips for R package developers. :eyes:

### Roaringly Acknowledge Organizations with ROR IDs in DESCRIPTION

In our [recent blog post](/blog/2025/05/09/ror/) we explained what ROR IDs are: they are to organizations what ORCID are for individuals.

> They help acknowledge more precisely an organization (a company that owns the copyright, a non-profit that funded the work, etc.) in package's metadata.

Reviewer 2: ```suggestion
They help acknowledge more precisely an organization (a company that owns the copyright, a non-profit that funded the work, etc.) in the package's metadata.
```

[Read more](/blog/2025/05/09/ror/).

### Safeguard your documentation in check environments

Or, in other words, how to have your package *not* fail `R CMD check` on CRAN, but examples and vignettes still rendered in pkgdown documentation?

> Thanks to Alasdair Warwick, maintainer of the recently reviewed [gtexr package](https://docs.ropensci.org/gtexr/), for summarizing [some advice](https://github.com/ropensci-org/rotemplate/issues/120#issuecomment-2831521763) that we copy below:

Reviewer 2: ```suggestion
Thanks to Alasdair Warwick, maintainer of the recently reviewed [gtexr package](https://docs.ropensci.org/gtexr/), for summarizing [some advice](https://github.com/ropensci-org/rotemplate/issues/120#issuecomment-2831521763) that we share here:
```

- For function examples, use the  [`examplesIf` roxygen2 tag](https://roxygen2.r-lib.org/articles/rd.html?q=examplesIf#examples) with the `IN_PKGDOWN` variable e.g. `#' @examplesIf identical(Sys.getenv("IN_PKGDOWN"), "true")`

> - For vignettes, either pre-build if special tools/data/credentials are required which are unavailable on generic build servers (see https://ropensci.org/blog/2019/12/08/precompute-vignettes/), or use the `IN_PKGDOWN` variable with knitr eval option e.g.

Reviewer 2: So cool! I have got to get that Dev Corner summary blog post up and running!

```r
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval = Sys.getenv("IN_PKGDOWN") == "true"
)
```

Examples:

- `examplesIf`:  https://github.com/ropensci/gtexr/blob/592ac781672f07eb67e935d4155570c5960d1fdb/R/get_service_info.R#L14 (see also Docs of the roxygen2 tag: https://roxygen2.r-lib.org/articles/rd.html?q=examplesIf#examples)
- Vignette, knitr eval option:   https://github.com/ropensci/gtexr/blob/592ac781672f07eb67e935d4155570c5960d1fdb/vignettes/gtexr.Rmd#L16

### More on {renv} and R versions

To follow up on last newsletter's item ["🔗
CI: Pin the R version if using renv"](/blog/2025/04/28/news-april-2025/#ci-pin-the-r-version-if-using-renv), a tip by Hugo Gruson:
it's also possible to automatically set the R version to the one documented in `renv.lock`
when using the [`r-lib/setup-r`action](https://github.com/r-lib/actions/blob/4f58f8ffa872cbd4cb016c51ec497fdbe0a02711/setup-r/README.md?plain=1#L37).

> ### `usethis::use_air()`

Reviewer 2: Because code is highlighted in pink, I was confused at first as to whether this was a heading or related to the previous section. Perhaps consider this?

```suggestion
### Air gets a usethis function: `usethis::use_air()`
```
